PyInstaller/building/splash.py

```diff
@@ -0,0 +1,291 @@
+            logger.info("Building %s because file %s changed",
+                        self.tocbasename, self.bitmap_file)
+            return True
+        return False
+
+    def _check_txt_rect(self):
+        if len(self.text_rect) != 4:
+            raise ValueError("wnd_text_rect must be a tuple of four positive integers!")
+        if any([not (isinstance(a, int) and a > 0) for a in self.text_rect]):
+            raise ValueError("wnd_text_rect must be a tuple of four positive integers!")
+
+    def assemble(self):
+        logger.info("Building Splash %s", self.name)
+
+        with open(self.bitmap_file, 'rb') as file:
+            # analyse bitmap file header
+            file_header = BITMAPFILEHEADER()
+            file_header.fromfile(file)
+
+            # Check if the bitmap magic number is correct
+            if not file_header.bfType == b"BM":
+                raise SystemExit('Supplied file is not an valid bitmap.')
+
+            # Load bitmap info from file
+            info_header = BITMAPINFOHEADER()
+            info_header.fromfile(file)
+
+            # Skip data from extended headers
+            file.seek(file_header.bfOffBits)
+
+            # Total number of pixels in the bitmap
+            image_size = info_header.biWidth * info_header.biHeight
+
+            # Only bitmaps with 32 bit and BI_BITFIELD
+            # or with 24 bit and BI_RGB are supported.
+            transparency = False
+            bitmap_type_info = (info_header.biBitCount, info_header.biCompression)
+            if bitmap_type_info == (32, 3):  # BitCount == 32 and Compression == BI_BITFIELDS
+                transparency = True
+                # Under Windows, at 32 bit the pixels have to be
+                # pre-multiplied with their alpha values.
+                if is_win:
+                    logger.info("Using 32bit Bitmap, pre-multiply color information.")
+                    bitmap_bits = _win_pre_multiply(file, image_size)
+
+                else:
+                    color_struct = struct.calcsize("<BBBB")
+                    bitmap_bits = file.read(image_size * color_struct)
+
+            elif bitmap_type_info == (24, 0):  # BitCount == 24 and Compression == BI_RGB
+                # RGBTRIPLE structure
+                color_struct = struct.calcsize("<BBB")
+                bitmap_bits = file.read(image_size * color_struct)
+
+            else:
+                raise SystemExit("Bitmap file format is not supported. Please refer to the documentation.")
+
+        splash_res = SplashResourceWriter(wnd_width=info_header.biWidth,
+                                          wnd_height=info_header.biHeight,
+                                          wnd_txt_rect=self.text_rect,
+                                          wnd_txt_clr=self.text_color,
+                                          transparency=transparency,
+                                          bitmap=bitmap_bits)
+        splash_res.write_to_file(self.name)
```

Comment:
Okay, maybe one should check the line length in this code again and correct it if necessary (at least Travis/Flake8 is raising the alarm here)